contracts/SponsorFaucet.sol

```diff
@@ -3,90 +3,62 @@ pragma experimental ABIEncoderV2;
 import "@openzeppelin/contracts/lifecycle/Pausable.sol";
+import "@openzeppelin/contracts/math/SafeMath.sol";
 import "./Lib/ReentrancyGuard.sol";
-
-contract SponsorWhitelistControl {
-    // ------------------------------------------------------------------------
-    // Someone will sponsor the gas cost for contract `contract_addr` with an
-    // `upper_bound` for a single transaction.
-    // ------------------------------------------------------------------------
-    function set_sponsor_for_gas(address contract_addr, uint upper_bound) public payable {
-    }
-
-    // ------------------------------------------------------------------------
-    // Someone will sponsor the storage collateral for contract `contract_addr`.
-    // ------------------------------------------------------------------------
-    function set_sponsor_for_collateral(address contract_addr) public payable {
-    }
-
-    // ------------------------------------------------------------------------
-    // Add commission privilege for address `user` to some contract.
-    // ------------------------------------------------------------------------
-    function add_privilege(address[] memory) public {
-    }
-
-    // ------------------------------------------------------------------------
-    // Remove commission privilege for address `user` from some contract.
-    // ------------------------------------------------------------------------
-    function remove_privilege(address[] memory) public {
-    }
-}
+import "./InternalContract.sol";
 
 contract SponsorFaucet is Ownable, Pausable, ReentrancyGuard {
+    using SafeMath for uint256;
     struct detail {
         uint256 cnt;
         uint256 limit;
     }
 
     uint256 public gas_bound;
-    uint256 public storage_bound;
+    uint256 public collateral_bound;
     mapping(address=>detail) public dapps;
-    mapping(address=>uint256) public sponsors;
     
-    event applied(address dapp);
-    event receieved(address sponsor, uint256 amount);
-
+    event applied(address indexed dapp);
+    
     SponsorWhitelistControl cpc = SponsorWhitelistControl(0x0888000000000000000000000000000000000001);
     
-    constructor(uint256 bound, uint256 s_bound) public {
-        gas_bound = bound;
-        storage_bound = s_bound;
+    constructor(uint256 gasBound, uint256 collateralBound) public {
+        gas_bound = gasBound;
+        collateral_bound = collateralBound;
     }
     
     /*** Dapp dev calls ***/ 
     //apply cfx for gas & storage
-    function applyFor(address dapp) public onlyOwner nonReentrant whenNotPaused {
-        cpc.set_sponsor_for_gas.value(gas_bound)(dapp, gas_bound);
-        cpc.set_sponsor_for_collateral.value(storage_bound)(dapp);
-        dapps[dapp].cnt++;
-        dapps[dapp].limit += (gas_bound+storage_bound);
+    function applyFor(address dapp) public nonReentrant whenNotPaused {
+        cpc.set_sponsor_for_gas.value(gas_bound)(dapp, gas_bound.div(1000));
```

Comment:
why use `gas_bound.div(1000)` as gas fee upper bound?

Reply:
set_sponsor_for_gas requires:
**The number of transferred tokens should be at least 1000 times of the upper_bound.**
so simply div(1000)

Reply:
This is a minimum value, we should sponsor more transactions. So, just let admin to set the `upper_bound` for gas fee.

Reply:
upper_bound added